src/daft-parquet/src/file.rs

```diff
@@ -371,9 +380,136 @@ impl ParquetFileReader {
 
     pub async fn read_from_ranges_into_table_stream(
         self,
-        _ranges: Arc<RangesContainer>,
-    ) -> BoxStream<'static, DaftResult<Table>> {
-        todo!("Implement streaming reads for remote parquet files")
+        ranges: Arc<RangesContainer>,
+        maintain_order: bool,
+        predicate: Option<ExprRef>,
+        original_columns: Option<Vec<String>>,
+        original_num_rows: Option<usize>,
+    ) -> DaftResult<BoxStream<'static, DaftResult<Table>>> {
+        let rt_handle = tokio::runtime::Handle::current();
+        let daft_schema = daft_core::schema::Schema::try_from(self.arrow_schema.as_ref())?;
+        let schema_ref = Arc::new(daft_schema);
+
+        let mut table_iters = Vec::with_capacity(self.row_ranges.len());
+        for row_range in self.row_ranges.iter() {
+            let rg = self
+                .metadata
+                .row_groups
+                .get(row_range.row_group_index)
+                .expect("Row Group index should be in bounds");
+            let num_rows = rg.num_rows().min(row_range.start + row_range.num_rows);
+            let columns = rg.columns();
+
+            let mut arr_iters = Vec::with_capacity(self.arrow_schema.fields.len());
+            for field in self.arrow_schema.fields.iter() {
+                let filtered_cols_idx = columns
+                    .iter()
+                    .enumerate()
+                    .filter(|(_, x)| x.descriptor().path_in_schema[0] == field.name)
+                    .map(|(i, _)| i)
+                    .collect::<Vec<_>>();
+
+                let needed_byte_ranges = filtered_cols_idx
+                    .iter()
+                    .map(|i| {
+                        let c = columns.get(*i).unwrap();
+                        let (start, len) = c.byte_range();
+                        let end: u64 = start + len;
+                        start as usize..end as usize
+                    })
+                    .collect::<Vec<_>>();
+
+                let mut range_readers = Vec::with_capacity(filtered_cols_idx.len());
+                for range in needed_byte_ranges.into_iter() {
```

Comment:
instead of collecting `range_readers` into a `Vec`, you can probably do this in 1 loop, producing `decompressed_iters`, `ptypes`, `num_values`, etc